aws_lambda_script/Dockerfile

```diff
@@ -0,0 +1,27 @@
+FROM python:3.12-slim
+
+# Create a non-root user and group
+RUN groupadd -r appuser && useradd -r -g appuser appuser
+
+# Install AWS Lambda Runtime Interface Client
+RUN pip install --no-cache-dir awslambdaric
+
+# Set working directory
+WORKDIR /app
+
+# Copy requirements and install dependencies
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r requirements.txt
```

Comment:
[Kieran uses pip ](https://github.com/ONS-Innovation/github-policy-dashboard/blob/main/lambda_data_logger/Dockerfile) in Docker even though the repo uses Poetry.

I think it's more complex to use Poetry within Dockerfile but I could. What do you think @TotalDwarf03?

Reply:
Since this repository is just for the lambda, you could use either. PIP is used for the other lambdas because poetry is used for the main part of the tool (i.e PIP for lambda, Poetry for Dashboard). Another option which we could do (up for discussion) is switch to full poetry for projects, using dependency groups to denote which is for which (this is probably the best approach and is tech debt).

I don't mind either package manager here though. A positive of using pip is that you don't have to install poetry within the docker image, meaning smaller image and quicker push times.

Reply:
I've decided to move the whole repo to PIP. It felt like there was no point in using Poetry for such a small repository. It's just one file with the Lambda code in. I also ran into a tonne of problems with containerising using Lambda then trying to run locally and in the Lambda. As in, I could containerise, push to ECR, deploy new image on Lambda, but the running locally wouldn't work.

I've made it so the user can build then run the container locally, but as it's a function, nothing is calling it. I need to work out how to do the `curl` command to trigger it.

Reply:
You can `curl` and it works. Main issue was using `python:3.12-slim` instead of `public.ecr.aws/lambda/python:3.12`.